File: core/src/test/java/com/mower/domain/CardinalPointShould.java
package com.mower.domain;

import org.junit.jupiter.api.Test;

import static com.mower.domain.CardinalPoint.EAST;
import static com.mower.domain.CardinalPoint.NORTH;
import static com.mower.domain.CardinalPoint.SOUTH;
import static com.mower.domain.CardinalPoint.WEST;
import static org.junit.jupiter.api.Assertions.assertSame;
import static org.junit.jupiter.api.Assertions.assertThrows;

class CardinalPointShould {

  private static final String UNKNOWN_CARDINAL_POINT_CODE = "UNKNOWN CARDINAL POINT CODE";

  @Test
  void throwUnknownCardinalPointCodeIfCardinalPointCodeIsUnknown() {
    assertThrows(UnknownCardinalPointCode.class, () -> CardinalPoint.fromCode(UNKNOWN_CARDINAL_POINT_CODE));
  }

  @Test
  void returnNorthFromNCode() {
    assertSame(NORTH, CardinalPoint.fromCode("N"));
  }

  @Test
  void returnEastFromECode() {
    assertSame(EAST, CardinalPoint.fromCode("E"));
  }

  @Test
  void returnSouthFromSCode() {
    assertSame(SOUTH, CardinalPoint.fromCode("S"));
  }

  @Test
  void returnWestFromWCode() {
    assertSame(WEST, CardinalPoint.fromCode("W"));
  }

  @Test
  void returnNorthCode() {
    assertSame("N", NORTH.code());
  }

  @Test
  void returnEastCode() {
    assertSame("E", EAST.code());
  }

  @Test
  void returnSouthCode() {
    assertSame("S", SOUTH.code());
  }

  @Test
  void returnWestCode() {
    assertSame("W", WEST.code());
  }

}

File: core/src/test/kotlin/com/mower/domain/CardinalPointShould.kt
package com.mower.domain

import com.mower.domain.CardinalPoint.Companion.fromCode
import com.mower.domain.exception.UnknownCardinalPointCode
import org.junit.jupiter.api.Test
import kotlin.test.assertFailsWith
import kotlin.test.assertSame

class CardinalPointShould {
    private val UNKNOWN_CARDINAL_POINT_CODE = "UNKNOWN CARDINAL POINT CODE"

    @Test
    fun throwUnknownCardinalPointCodeIfCardinalPointCodeIsUnknown() {
        assertFailsWith<UnknownCardinalPointCode> {
            fromCode(UNKNOWN_CARDINAL_POINT_CODE)
        }
    }

    @Test
    fun returnNorthFromNCode() {
        assertSame(CardinalPoint.NORTH, fromCode("N"))
    }

    @Test
    fun returnEastFromECode() {
        assertSame(CardinalPoint.EAST, fromCode("E"))
    }

    @Test
    fun returnSouthFromSCode() {
        assertSame(CardinalPoint.SOUTH, fromCode("S"))
    }

    @Test
    fun returnWestFromWCode() {
        assertSame(CardinalPoint.WEST, fromCode("W"))
    }

    @Test
    fun returnNorthCode() {
        assertSame("N", CardinalPoint.NORTH.code)
    }

    @Test
    fun returnEastCode() {
        assertSame("E", CardinalPoint.EAST.code)
    }

    @Test
    fun returnSouthCode() {
        assertSame("S", CardinalPoint.SOUTH.code)
    }

    @Test
    fun returnWestCode() {
        assertSame("W", CardinalPoint.WEST.code)
    }
}
